### PR TITLE
auth fixes for new pvw, better handling of login promises

### PR DESCRIPTION
**Pull request by TristanWright**
I was running into a lot of auth problems from a fresh vm when testing the HPCC-deploy demo branch.

This branch fixes a number of registration and login issues I was experiencing.

### CODE REVIEW

**Comment by TristanWright** [comment]
I believe it fixes fix #309, maybe #220?

File: src/config/routes.js [hunk 1]
```diff
@@ -46,9 +46,11 @@ function redirectToHome(nextState, replace) {
...
   }
-  client.isLoggedIn().then(() => {
-    replace('/');
-  });
+  client.isLoggedIn()
+    .then(() => {
+      replace('/');
+    })
+    .catch(() => {});
 }
```
Comment (cjh1) [inline src/config/routes.js]: Does this just ignore errors?

**Comment by cjh1** [comment]
This branch resolves the auth issues I was seeing.

File: src/config/routes.js (hunk 1 above)
Comment (TristanWright) [inline src/config/routes.js]: Hmm we'd get an `unhandled error` I guess both should redirect to '/' looking into it and then I'm going to merge

**Pull request merged**